auth-web/src/components/auth/manage-business/AddNameRequestFormOld.vue

```diff
@@ -0,0 +1,250 @@
+    v => this.isValidateEmail(v) || 'Email is Invalid'
+  ]
+
+  VALID_NR_FORMAT = new RegExp(/^(NR)?\s*(\d{7})$/)
+
+  nrNumber = ''
+  nrNumberKey = 0
+  applicantPhoneNumber = ''
+  applicantEmail = ''
+  isLoading = false
+
+  formatNrNumber (value): string {
+    let formattedNrNumber = value?.toUpperCase()
+    if (this.VALID_NR_FORMAT.test(formattedNrNumber)) {
+      formattedNrNumber = 'NR ' + this.VALID_NR_FORMAT.exec(formattedNrNumber)[2]
+    }
+    // Force a re-render when the content is the same (reactivity issue)
+    if (formattedNrNumber === this.nrNumber) {
+      this.nrNumberKey++
+    }
+    return formattedNrNumber
+  }
+
+  $refs: {
+    addNRForm: HTMLFormElement,
+    nrNumber: HTMLFormElement,
+    helpDialog: HelpDialog
+  }
+
+  isFormValid (): boolean {
+    return !!this.nrNumber &&
+      (!!this.applicantPhoneNumber || !!this.applicantEmail) &&
+      this.isValidNrNumber(this.nrNumber)
+  }
+
+  isInputEntered (value: any, inputType: string): boolean {
+    return (!!((inputType === 'email')
+      ? this.applicantPhoneNumber
+      : this.applicantEmail) || !!value)
+  }
+
+  isValidateEmail (value: any): boolean {
+    return ((!!this.applicantPhoneNumber && !!value) ||
+      !!CommonUtils.validateEmailFormat(value))
+  }
+
+  isValidNrNumber (value: any): boolean {
+    return this.VALID_NR_FORMAT.test(value)
+  }
+
+  async add (): Promise<void> {
+    if (this.isFormValid()) {
+      this.isLoading = true
+      try {
+        // attempt to add business
+        const nrResponse = await this.addNameRequest({
+          businessIdentifier: this.nrNumber,
+          phone: this.applicantPhoneNumber.replace(/-/g, ''),
+          email: this.applicantEmail
+        })
+        if (nrResponse?.status === 201) {
+          // emit event to let parent know business added
+          this.$emit('add-success')
+        } else {
+          this.$emit('add-unknown-error')
+        }
+      } catch (exception) {
+        if (exception.response?.status === StatusCodes.BAD_REQUEST) {
+          this.$emit('add-failed-show-msg', exception.response?.data?.message || '')
+        } else if (exception.response?.status === StatusCodes.NOT_FOUND) {
+          this.$emit('add-failed-no-entity')
+        } else {
+          this.$emit('add-unknown-error')
+        }
+      } finally {
+        this.resetForm('')
+      }
+    }
+  }
+
+  resetForm (event: string): void {
+    this.nrNumber = ''
+    this.applicantEmail = ''
+    this.applicantPhoneNumber = ''
+    this.$refs.addNRForm.resetValidation()
+    this.isLoading = false
+    if (event) {
+      this.$emit('on-cancel')
+    }
+  }
+
+  openHelp (): void {
+    this.$refs.helpDialog.open()
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '$assets/scss/theme.scss';
+
+.form__btns {
+  display: flex;
+  justify-content: flex-end;
+
+  .v-btn + .v-btn {
+    margin-left: 0.5rem;
+  }
+
+  .v-btn[data-test="cancel-button"],
```

Comment:
Probably shouldn't be binding by data-test - I think we should look for ID in this case